src/components/Dashboard/Header/Authentication/Signup.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import {
+  Form, Button, Container, Row, Col,
+} from 'react-bootstrap';
+import { Redirect } from 'react-router-dom';
+import { Toaster, Intent } from '@blueprintjs/core';
+import db from '../../../../fire';
+import NavBar from '../Header';
+
+export default class Signup extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redirect: false,
+    };
+  }
+
+  createAccount = (event) => {
+    event.preventDefault();
+    const email = this.emailInput.value;
+    const password = this.passwordInput.value;
+
+    if (this.validate(event)) {
+      console.log(event);
+      db.auth().fetchProvidersForEmail(email)
+        .then((providers) => {
+          console.log(providers);
+          if (providers.length === 0) {
+            // catch user
+            return db.auth().createUserWithEmailAndPassword(email, password);
+          } else {
+            this.loginForm.reset();
+            this.triggerValidation(this.emailInput, 'That email is taken. Try another.');
+          }
+        })
+        .then((user) => {
+          if (user && user.user && user.user.email) {
+            this.loginForm.reset();
+            this.setState({ redirect: true });
+          }
+        })
+        .catch((error) => {
+          this.toaster.show({ intent: Intent.DANGER, message: error.message });
+        });
+    }
+  }
+
+  validate = () => {
+    const name = this.nameInput.value;
+    const email = this.emailInput.value;
+    const password = this.passwordInput.value;
+    const confirmPassword = this.confirmPasswordInput.value;
+    let valid = true;
```

Comment:
you can use construction like this to avoid complicated if-else and code will stop executing after first return

```
 if (!name) {
      this.triggerValidation(this.nameInput, 'Name is required.')
      return false;
    } 
    if (!email) {
     ...
    }
    return true;
```